assign-pull-requests.py

```diff
@@ -125,6 +125,33 @@ def main(ref_repo_path):
 
     return 0
 
+# check for GCO sign-off
+RE_SIGNOFFBY = re.compile(r'^signed-off-by:\s+', re.I)
+def commit_contains_correct_signoff(commit):
+    """ Given a commit, it is considered to contain a correct sign-off IFF:
+    It has a footer line starting with 'Signed-off-by: '
+    AND that line contains the email address of the committer.
+
+    Other sign-off lines MAY also be present, but do not affect the outcome of
+    this check. It's entirely valid for one person to have multiple sign-off
+    lines, using different email addresses (signing off on something for both
+    work and Gentoo development)
+    """
+    lines = commit.message.splitlines()
+    committer_email = commit.committer.email.lower() # laziness here
+    # Get all lines that start with signed-off-by
+    signofflines = [x for x in lines if RE_SIGNOFFBY.match(x)]
+    for line in signofflines:
+        # Split the line, grab everything with '@' in it
+        possible_emails = [word for word in re.split(r'\s+', line) if '@' in word]
```

Comment:
Do we really want to hack it like this? I'd say using [email.utils.parseaddr](https://docs.python.org/3/library/email.utils.html#email.utils.parseaddr) would be better.